pkg/trie/triedb/node.go

```diff
@@ -0,0 +1,195 @@
+// Copyright 2024 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package triedb
+
+import (
+	"bytes"
+
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/pkg/trie/db"
+	"github.com/ChainSafe/gossamer/pkg/trie/triedb/codec"
+)
+
+type StorageHandle struct{ int }
+
+func (sh StorageHandle) toNodeHandle() NodeHandle {
+	return InMemory{idx: sh}
+}
+
+type NodeHandle interface {
+	isNodeHandle()
+}
+
+type (
+	InMemory struct {
+		idx StorageHandle
+	}
+	Hash struct {
+		hash common.Hash
+	}
+)
+
+func (InMemory) isNodeHandle() {}
```

Comment:
These are here just to adhere to the interface?

Reply:
Yes, and using them as an enum